### PR TITLE
service/dap: add go build stderr to error response

### PR DESCRIPTION
**Pull request by suzmue**
When there is a build error, the error returned by the go command includes the exit code,
but not much detail. This change adds the information from stderr to the error message
to provide more useful information to the client.

Updates golang/vscode-go#1425

### CODE REVIEW

**Comment by suzmue** [comment]
cc @hyangah @polinasok

File: service/dap/server_test.go [hunk 1]
```diff
@@ -3080,18 +3080,18 @@ func TestBadLaunchRequests(t *testing.T) {
...
 
 		client.LaunchRequest("exec", fixture.Source, stopOnEntry)
 		expectFailedToLaunch(client.ExpectErrorResponse(t)) // Not an executable
 
-		client.LaunchRequestWithArgs(map[string]interface{}{"mode": "debug", "program": fixture.Source, "buildFlags": "bad flags"})
-		expectFailedToLaunchWithMessage(client.ExpectErrorResponse(t), "Failed to launch: Build error: exit status 1")
-		client.LaunchRequestWithArgs(map[string]interface{}{"mode": "debug", "program": fixture.Source, "noDebug": true, "buildFlags": "bad flags"})
-		expectFailedToLaunchWithMessage(client.ExpectErrorResponse(t), "Failed to launch: Build error: exit status 1")
+		client.LaunchRequestWithArgs(map[string]interface{}{"mode": "debug", "program": fixture.Source, "buildFlags": "-bad -flags"})
+		expectFailedToLaunchWithMessage(client.ExpectErrorResponse(t), "Failed to launch: Build error: flag provided but not defined: -bad\nusage: go build [-o output] [build flags] [packages]\nRun 'go help build' for details. (exit status 2)")
```
Comment (aarzilli) [inline service/dap/server_test.go:3092]: This test is too precise and breaks for older versions of go (see teamcity failure). Just check that the message contains "flag provided but not defined", or something like that.

Comment (suzmue) [reply service/dap/server_test.go]: Done!

**Review by aarzilli — APPROVED** [review]
LGTM

**Pull request merged**